services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -574,6 +577,33 @@ public VeniceHelixAdmin(
         realTimeTopicSwitcher,
         accessController,
         helixAdminClient);
+
+    for (String clusterName: multiClusterConfigs.getClusters()) {
+      HelixLiveInstanceMonitor liveInstanceMonitor = new HelixLiveInstanceMonitor(this.zkClient, clusterName);
+      // Register new instance callback
+      liveInstanceMonitor.registerLiveInstanceChangedListener(new LiveInstanceChangedListener() {
+        @Override
+        public void handleNewInstances(Set<Instance> newInstances) {
+          long startTime = System.currentTimeMillis();
+          for (Instance instance: newInstances) {
+            Map<String, List<String>> disabledPartitions =
+                helixAdminClient.getDisabledPartitionsMap(clusterName, instance.getNodeId());
+            for (Map.Entry<String, List<String>> entry: disabledPartitions.entrySet()) {
+              helixAdminClient
+                  .enablePartition(true, clusterName, instance.getNodeId(), entry.getKey(), entry.getValue());
+            }
+          }
+          LOGGER.info(
+              "Enabling disabled replicas for instances {} took {} ",
```

Comment:
If we are logging, I think it might be worth to collect a list/array of enabled replicas.

Reply:
Its is already collecting the object and print them in 1 line?

Reply:
I wasn't talking about just the instance IDs. Since disabling and enabling of replicas shouldn't be very common. If we are logging it already we probably want all the information. i.e. instance id plus the associated store version partition that was re-enabled.